descabi-morello/descabi-morello.rst

```diff
@@ -0,0 +1,769 @@
+
+Change history
+^^^^^^^^^^^^^^
+
+.. class:: morello-desc-change
+
+.. table::
+
+  +---------------+--------------------+-----------------------------------------+
+  | Issue         | Date               | Change                                  |
+  +===============+====================+=========================================+
+  | 00alpha       | 7th June 2021      | Document released on GitHub.            |
+  +---------------+--------------------+-----------------------------------------+
+
+References
+----------
+
+This document refers to, or is referred to by, the following documents.
+
+.. class:: morellodesc-references
+
+.. table::
+
+  +-------------------+----------------------------+------------------------------------------------+
+  | Ref               | External reference or URL  | Title                                          |
+  +===================+============================+================================================+
+  | MORELLO_DESC_     | This document              | Morello Descriptor ABI for the Arm 64-bit      |
+  |                   |                            | Architecture (AArch64).                        |
+  +-------------------+----------------------------+------------------------------------------------+
+  | MORELLO_ELF_      | |morello-elf-url|          | ELF for the Arm 64-bit Architecture (AArch64). |
+  +-------------------+----------------------------+------------------------------------------------+
+  | MORELLO_PCS_      | |morello-pcs-url|          | ELF for the Arm 64-bit Architecture (AArch64). |
+  +-------------------+----------------------------+------------------------------------------------+
+  | MORELLO_ARM_      | |morello-arm-url|          | Arm Architecture Reference Manual Supplement   |
+  |                   |                            | Morello for A-profile Architecture.            |
+  +-------------------+----------------------------+------------------------------------------------+
+
+Terms and abbreviations
+-----------------------
+
+The ABI for the Morello extensions to the Arm 64-bit Architecture uses the
+following terms and abbreviations.
+
+C64
+  The instruction set available when the Morello extensions are used.
+
+A64
+  The instruction set available when in AArch64 state.
+
+Other terms may be defined when first used.
+
+.. raw:: pdf
+
+   PageBreak
+
+About This Specification
+========================
+
+This specification provides the Morello Descriptor ABI specification for the
+Arm 64-bit Architecture (AArch64), and is expected to be used along with MORELLO_ELF_
+and MORELLO_PCS_.
+
+Morello Descriptor ABI
+======================
+
+The Morello Descriptor ABI is a variant of the pure capability ABI which decouples code and
+global data addresses, allowing the same code to be used with multiple instances of global data.
+This allows compartments to share code in the same address space, while having separata data
```

Comment:
Yes, there is an assumtion that CCTLR.ADRDPB should be set. I'll make that explicit.

Different compartments should have different CTPIDR_ELx/c28/global data so that should produce different values per compartment for the same TLS symbol.

The TLS details are no really covered in this though, I'll send an update for that (there should be a small deviation for purecap when the GD resolver needs to be called).